go/action_kit_sdk/prepare_helper/prepare_helper.go

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023 Steadybit GmbH
+
+package prepare_helper
+
+import (
+	"github.com/steadybit/action-kit/go/action_kit_api/v2"
+	extension_kit "github.com/steadybit/extension-kit"
+	"github.com/steadybit/extension-kit/extconversion"
+)
+
+type PrepareResultModifier func(result *action_kit_api.PrepareResult)
+
+func WithArtifacts(artifacts *action_kit_api.Artifacts) PrepareResultModifier {
+	return func(result *action_kit_api.PrepareResult) {
+		result.Artifacts = artifacts
+	}
+}
+func WithError(error *action_kit_api.ActionKitError) PrepareResultModifier {
+	return func(result *action_kit_api.PrepareResult) {
+		result.Error = error
+	}
+}
+func WithMessages(messages *action_kit_api.Messages) PrepareResultModifier {
+	return func(result *action_kit_api.PrepareResult) {
+		result.Messages = messages
+	}
+}
+func WithMetrics(metrics *action_kit_api.Metrics) PrepareResultModifier {
+	return func(result *action_kit_api.PrepareResult) {
+		result.Metrics = metrics
+	}
+}
+
+func NewPrepareResult(state interface{}, modifier ...PrepareResultModifier) (*action_kit_api.PrepareResult, error) {
```

Comment:
I'm a bit torn on this options pattern...
yes it is very flexible and extensible but also maybe not that approachable....


We could also have single parameter of type
```
package action_kit_sdk;

type PrepareResult {
	State interface{}
	Artifacts *Artifacts
	Error *ActionKitError
	Messages *Messages
	Metrics  *Metrics 
}
```

that will be converted into a action_kit_api.PrepareResult. (note the different packages).

so that would be a bit different in usage:

```
status_helper.NewPrepareResult(status_helper.PrepareResult{
		State: state,
		Messages: &action_kit_api.Messages{
			{Level: extutil.Ptr(action_kit_api.Info), Message: "This is a test Message from Status"},
		},
	})
```

vs.


```
return status_helper.NewStatusResult(state,
 		status_helper.WithMessages(&action_kit_api.Messages{
 			{Level: extutil.Ptr(action_kit_api.Info), Message: "This is a test Message from Status"},
 		})
 	)
 ```
 
@ReuDa @bripkens you have some opinion on that??

Reply:
Agree that this pattern might be a bit unwieldy / too complicated to grok. 

What if instead we add an additional return type to prepare which is the state object? We would then encode it and set it on the prepare result in the SDK.

Reply:
I removed all helper-packages. We pass the converted state into the methods as a pointer.